libqtile/widget/base.py

```diff
@@ -369,6 +370,35 @@ def mouse_enter(self, x, y):
     def mouse_leave(self, x, y):
         pass
 
+    def _draw_with_mirrors(self):
+        self._old_draw()
+        for mirror in self.mirrors:
+            if not mirror.configured:
+                continue
+
+            if mirror.length_type == bar.CALCULATED:
+                mirror.bar.draw()
```

Comment:
Could this not result in an infinite loop between bar.draw and mirror.draw?

Reply:
Hmm. It could if they were on the same bar...

Reply:
Strange edge case but what if there were 2 widgets, and each was passed to 2 bars, but the mirrors are on different bars. Then they will induce recursion in each other via their mirrors. Maybe this is impossible because one bar is guaranteed to be setup before the other and get all of the non-mirrored widgets, and that also prevents this happening when playing around with adding/removing screens etc. That sounds about right, right?

Reply:
Correct. Bars are initialised one at a time so I can't see a situation where Bar A has a mirror of a widget on Bar B, and Bar B has a mirror of a widget on Bar A.